l10n_nl_tax_declaration_reporting/report/report_vat.py

```diff
@@ -0,0 +1,288 @@
+                data['form']['period_to'])
+            periods_l = period_obj.read(
+                self.cr, self.uid, self.period_ids, ['name'])
+            for period in periods_l:
+                if res['periods'] == '':
+                    res['periods'] = period['name']
+                else:
+                    res['periods'] += ", " + period['name']
+
+        self._get_lines(data['form']['based_on'], data['form']['company_id'])
+        self._5AB = int(self._1AB + self._1BB + self._1CB +
+                        self._1DB + self._2AB + self._4AB + self._4BB)
+        self._5CB = int(self._1AB + self._1BB + self._1CB +
+                        self._1DB + self._2AB + self._4AB +
+                        self._4BB - self._5BB)
+
+        docargs = {
+            'doc_ids': self._ids,
+            'doc_model': report.model,
+            'docs': self,
+            'time_now': fields.datetime.now(),
+            'chart': _chart_tax_name,
+            'fiscalyear': _fiscalyear_name,
+            'start_period': _period_from_name,
+            'end_period': _period_to_name,
+            'basedon': data['form']['based_on'],
+            '_1AO': int(self._1AO),
+            '_1AB': int(self._1AB),
+            '_1BO': int(self._1BO),
+            '_1BB': int(self._1BB),
+            '_1CO': int(self._1AO),
+            '_1CB': int(self._1CB),
+            '_1DO': int(self._1DO),
+            '_1DB': int(self._1DB),
+            '_2AO': int(self._2AO),
+            '_2AB': int(self._2AB),
+            '_3AO': int(self._3AO),
+            '_3BO': int(self._3BO),
+            '_3CO': int(self._3CO),
+            '_4AO': int(self._4AO),
+            '_4AB': int(self._4AB),
+            '_4BO': int(self._4BO),
+            '_4BB': int(self._4BB),
+            '_5AB': int(self._5AB),
+            '_5BB': int(self._5BB),
+            '_5CB': int(self._5CB),
+        }
+
+        return report_obj.render(
+            'l10n_nl_tax_declaration_reporting.tax_report_template', docargs)
+
+    def _get_lines(self, based_on, company_id=False,
+                   parent=False, level=0, context=None):
+        period_list = self.period_ids
+        res = self._get_codes(based_on, company_id, parent, level,
+                              period_list, context=context)
+        if period_list:
+            res = self._add_codes(based_on, res, period_list, context=context)
+        else:
+            self.cr.execute("select id from account_fiscalyear")
+            fy = self.cr.fetchall()
+            self.cr.execute(
+                "select id from account_period where fiscalyear_id = %s",
+                (fy[0][0],))
+            periods = self.cr.fetchall()
+            for p in periods:
+                period_list.append(p[0])
+            res = self._add_codes(based_on, res, period_list, context=context)
+
+        i = 0
+        while i < len(res):
+            # TODO refactor, e.g. using map or switch case?
```

Comment:
Please use a mapping, indeed. That allows you to raise properly when you encounter an unrecognized code.